packages/storage-web/src/index.ts

```diff
@@ -5,3 +5,96 @@
  * LICENSE file in the root directory of this source tree.
  *
  */
+import {
+  IStorageBackend,
+  EmptyStorageModel,
+  StorageOptions,
+} from '@react-native-community/async-storage';
+
+class WebStorage<T extends EmptyStorageModel = EmptyStorageModel>
+  implements IStorageBackend<T> {
+  storage: any;
+  constructor(sessionStorage: boolean | string = false) {
+    this.storage = sessionStorage ? window.sessionStorage : window.localStorage;
+  }
+
+  async getSingle<K extends keyof T>(
+    key: K,
+    opts?: StorageOptions,
+  ): Promise<T[K] | null> {
+    if (opts) {
+      // noop
+    }
+    return this.storage.getItem(key);
+  }
+
+  async setSingle<K extends keyof T>(
+    key: K,
+    value: T[K],
+    opts?: StorageOptions,
+  ): Promise<void> {
+    if (opts) {
+      // noop
+    }
+    return this.storage.setItem(key, value);
+  }
+
+  async getMany<K extends keyof T>(
+    keys: Array<K>,
+    opts?: StorageOptions,
+  ): Promise<{[k in K]: T[k] | null}> {
+    if (opts) {
+      // noop
+    }
+    return Promise.all(keys.map(k => ({[k]: this.storage.getItem(k)}))).then(
```

Comment:
@Krizzu I got this to return an object containing the key value pair, but I can't figure out how to get the types working correctly. Please let me know what you think. Thanks for all your help and patience so far!

Reply:
Sure thing.

Since this method (`getMany`) already returns a promise (because its marked as `async`), there's no need to return `Promise.all`. You can return the result of `reduce` directly.

So you'll eventually end up with something like this:

```typescript
   return keys.reduce(
      (storageValues, key) => {
        return {
          ...storageValues,
          [key]: this.storage.getItem(key),
        };
      },
      {} as {[k in K]: T[k] | null}, // here cast the initial value to the type we're going to return
    );
```

Reply:
Ah, this is much better! Thanks!

Reply:
Ok, I updated with your suggestion.